source/FreeRTOS_DHCPv6.c

```diff
@@ -934,112 +953,126 @@ static void prvSendDHCPMessage( NetworkEndPoint_t * pxEndPoint )
 
         /* Not useful, but MISRA issues a mandatory warning. */
         ( void ) memset( &( xMessage ), 0, sizeof( xMessage ) );
-        ( void ) xBitConfig_init( &( xMessage ), NULL, 256 ); /* Clear the message. */
 
-        switch( EP_DHCPData.eDHCPState )
+        if( xBitConfig_init( &( xMessage ), NULL, 256 ) == pdTRUE )

```

Comment:
1. The function xApplicationDHCPHook is being used for both IPv4 and IPv6 , hence this should be using IP_Address_t instead of the IPv4 address. Infact, we should be using ipconfigIPv4_BACKWARD_COMPATIBLE  to define 2 versions of this API as we have done in other callback functions.
2. Line 565 uses ipv4_defaults (( void ) memcpy( &( pxEndPoint->ipv4_settings ), &( pxEndPoint->ipv4_defaults ), sizeof( pxEndPoint->ipv4_settings ) );. This should be modified to corresponding ipv6 names. Line 468 does it correctly.

Reply:
> Can we use a macro instead

Using a macro in stead of what?

> this should be using IP_Address_t instead of the IPv4 address

Oops yes, there is some unfinished work there. Thank you, I will check that.

> Line 565 uses ipv4_defaults (( void ) memcpy( &( pxEndPoint->ipv4_settings ), &( pxEndPoint->ipv4_defaults ), 

will also check that.

Reply:
I'd suggest to separate xApplicationDHCPHook as another PR. And that doesn't impact DHCP functionality.